index.js

```diff
@@ -273,10 +279,11 @@ class NetFacility extends Base {
           await this.rpc.destroy()
         }
 
-        await this.dht.destroy()
-        if (this.lookup) {
-          await this.lookup.stop()
+        if (this.swarm && this.opts.swarmDestroyReq) {
```

Comment:
in what scenario will we have a `this.swarm` initialized, but we won't want to destroy it?